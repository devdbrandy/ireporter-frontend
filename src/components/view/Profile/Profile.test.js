import React from 'react';
import { shallow } from 'enzyme';
import { Profile } from './index';

/**
 * Wrapper for enzyme shallow component
 *
 * @returns {object} The mocked props and component
 */
function setup() {
  const props = {
    user: {},
    overview: {},
  };

  const component = shallow(<Profile {...props} />);
  return { props, component };
}

describe('<Profile />', () => {
  it('renders without crashing given the required props', () => {
    const component = setup();
    expect(component).toMatchSnapshot();
  });
});
