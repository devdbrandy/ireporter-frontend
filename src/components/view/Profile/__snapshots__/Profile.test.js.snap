// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Profile /> renders without crashing given the required props 1`] = `
Object {
  "component": <Fragment>
    <Layout>
      <withRouter(Panel)
        title="Profile"
      >
        <div
          className="user-profile row middle"
        >
          <div
            className="user--avatar column center"
          >
            <img
              alt="avatar"
            />
          </div>
          <div
            className="user--detail column center"
          >
            <h3>
               
            </h3>
            <Link
              className="btn btn-primary edit"
              to="/settings"
            >
              Edit Profile
            </Link>
            <hr
              className="dash"
            />
            <p
              className="user--bio"
            >
              No bio provided.
            </p>
          </div>
          <div
            className="user--stat column center text-center"
          >
            <div
              className="wrapper"
            >
              <span>
                Joined:
                 
                <i
                  className="user--registered"
                >
                  May 6th, 2019
                </i>
              </span>
              <hr
                className="dash"
              />
              <span>
                <i
                  className="incident-total"
                />
                 
                Incidents Published
              </span>
              <span>
                <i
                  className="incident-resolved"
                />
                 
                Incidents Resolved
              </span>
              <span>
                <i
                  className="incident-rejected"
                />
                 
                Incidents Rejected
              </span>
            </div>
          </div>
        </div>
      </withRouter(Panel)>
    </Layout>
  </Fragment>,
  "props": Object {
    "overview": Object {},
    "user": Object {},
  },
}
`;
